src/Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -99,7 +99,8 @@ public class BlockPuller : IBlockPuller
 
         /// <param name="blockHash">Hash of the delivered block.</param>
         /// <param name="block">The block.</param>
-        public delegate void OnBlockDownloadedCallback(uint256 blockHash, Block block);
+        /// <param name="peerId">The ID of the peer delivering the block.</param>
```

Comment:
The block was already delivered.
Maybe `Id of a peer that delivered the block`

Reply:
Done